### ISSUE TITLE
Provide config option to define default sort field and order

### ISSUE THREAD

**Issue by avezina** [Esri/solutions-components#308]
Right now, OBJECTID is used to sort the table and is always shown in the table. 

OBJECTID is a field that makes sense to the database but doesn't provide any useful context to the end users. Users may want to use another field to sort their data, such as the Request ID field below. 

In that case, users may not want to to see the OBJECTID field at all, and have a preset sorting set on the Request ID field
![image](https://github.com/Esri/solutions-components/assets/50338557/446a696b-39bf-40da-b0d4-6e2e1d4ea6e6)

cc @azizaparveen @chris-fox

**Comment by chris-fox** [comment]
@avezina, we definitely have the ability to configure which fields display in the table as something in the backlog for the app. This is an important requirement that we had previously in Crowdsource Manager.

With regard to setting a default sort field with the app loads, our initial thought was to have it behave like it did previously in Crowdsource Manager where the default sort order was OBJECTID descending. This was done to always have the most recent created features display at the top of the table. Users could still change the sort order at runtime, but the reverse sort on OBJECTID worked well for most cases and I am not aware of any user feedback requesting the ability to specify a different default sort order.

With this in mind I am inclined to have this same behavior for beta and wait for user feedback on if they require a different default sort order per layer.

**Comment by jmhauck** [comment]
Looked at the possibility of sorting based on OID as well as hiding the OID field and it does not appear that we can sort on a hidden field. The only way I saw it work was to add a small delay after the call to sort that would then hide the OID field. But if the OID field was marked not visible when creating the instance of the table or even immediately after the call to sort it would not work.

**Comment by azizaparveen** [comment]
We are providing config option for field order in R1 release, next release we will work on the default sort field.

**Comment by azizaparveen** [comment]
![image](https://github.com/Esri/solutions-components/assets/138827632/d68d4b30-3e52-482d-95fa-fc3e18f5b18b)

**Comment by azizaparveen** [comment]
add the objectid at the end of the table, and can we sort based on any field, add toggle for sort button

**Comment by azizaparveen** [comment]
@chris-fox @avezina, in today's meeting, we discussed about showing the sort button on hover, I am wondering if we can show the sort button always and mimic the table behavior. The table shows ascending and descending button by default,  once clicked it shows ascending icon at first, change the color of the icon and text to blue that gives a clear indication that particular field is used for sorting the table, clicked once more, it shows the descending arrow with text and icon color in blue, only thing I don't like is the icon but for consistency I am open to use the same icon.

![Image](https://github.com/Esri/solutions-components/assets/138827632/310e63ae-26c9-4c9f-a563-4ccafd0a70b1)

![Image](https://github.com/Esri/solutions-components/assets/138827632/8edfa15d-d014-4a96-8d21-bd4bd703a317)

**Comment by avezina** [comment]
@azizaparveen Makes sense to me. Not sure there would be a Default view though? There would always have to be a field defined for sorting - probably ObjectID by default?
Other icon options  https://developers.arcgis.com/calcite-design-system/icons/?query=sort
![image](https://github.com/Esri/solutions-components/assets/50338557/49801dba-7e2b-4fb9-9381-bbba6dde5c79)

**Comment by azizaparveen** [comment]
@avezina we now sort the table by ObjectID by default. 
These icons are also good alternatives, will send these to Lucy to get her opinion

**Comment by azizaparveen** [comment]
logged an issue in Template config repo
https://devtopia.esri.com/WebGIS/arcgis-template-configuration/issues/4447

**Comment by azizaparveen** [comment]
IA team suggested that we use dropdown to select default field and sort order, their concern with the our approach is the lack of clarity in indicating to users that clicking on the sorting icons establishes the default sorting field. Here is their suggestion.
![image](https://github.com/Esri/solutions-components/assets/138827632/a586ef9e-93fe-42e8-86ed-1681b159d5ac)

**Comment by sumitzarkar** [comment]
@azizaparveen 

This needs changes in Instant app for configuration, once those chanes are implemenetd we can implement it in the component

@jmhauck - please confirm

**Comment by jmhauck** [comment]
@sumitzarkar correct this will require changes from IA team in the repo listed above.

**Comment by azizaparveen** [comment]
@sumit, this is installed in devext, Interactivity tab ->Table Options -> clicking on the Fields button
![image](https://github.com/Esri/solutions-components/assets/138827632/2d5d16c3-413e-483e-a30c-e464b502e477)

**Comment by jmhauck** [comment]
Looks like this should already come through with `mapInfo.layerOptions` but I will confirm.

**Comment by prachichawande-ct** [comment]
@azizaparveen @avezina 

## Dev Checklist


Items ​ | Is required​ | Status | Comments​
-- | -- | -- | --
Required NLS changes?​ | No   |   | ​
Required Backward compatibility​ | Yes  |   | ​
Required 508​ | No  |   | ​
Need Sanitize?​ | No  |   | ​
___

## Impact Analysis Report

Impacted Areas | Comments
-- | --
Have Configuration changes?​ | No
Have Runtime changes?​ | Yes
Impact on RTL | No
Impact at runtime sorting feature functionality |  Features in the layer table should be sorted based on the configured sort field and sorting order
||For backward compatibility, by default features should be sort by OBJECTID field in descending order|

**Comment by sumitzarkar** [comment]
@jmhauck 

https://github.com/Esri/solutions-components/issues/308#issuecomment-2085812344

- Correct, we are getting this info through mapInfo. we don't need seperate props on manager component. 
- So we have removed this props from manager and updated the layer-tables to use the mapinfo to fetch the configured sortOrder and sortField

We will deliver this in the next code drop

**Comment by azizaparveen** [comment]
@sumitzarkar sorting is not working in the config preview but it works as expected at the runtime.

![Image](https://github.com/Esri/solutions-components/assets/138827632/8b3e5658-828e-429b-a268-c3b644effd6d)

**Comment by sumitzarkar** [comment]
@jmhauck 

>sorting is not working in the config preview but it works as expected at the runtime.

Please let us know if any changes required for this in manager component

**Comment by jmhauck** [comment]
I was having issues running the config app locally so I couldn't fully confirm but I did add a call to sort with mapInfo changes in layer-table.

**Comment by azizaparveen** [comment]
It looks good in devext.
